.circleci/config.yml

```diff
@@ -0,0 +1,65 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: circleci/node:12.20
+
+jobs:
+  lint:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+
+      - run: yarn
+      - run:
+          name: Run linter
+          command: yarn lint
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+  deploy:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run:
+          name: Publish package based on package.json version
+          command: npm publish
+
+workflows:
+  version: 2
+  lint:
+    jobs:
+      - lint
+
+  deploy:
+    jobs:
+      - lint:
```

Comment:
I think we can let the lint step perform at any push, any branch (this could lead to lint checks in our PR).

Reply:
Exactly ! This is why there is the `lint` workflow that isn't filtered at all
```
workflows:
  version: 2
  lint:
    jobs:
      - lint
```